x-pack/filebeat/input/salesforce/input.go

```diff
@@ -0,0 +1,506 @@
+
+			req.Header.Add("X-PrettyPrint", "1")
+
+			resp, err := s.sfdcConfig.Client.Do(req)
+			if err != nil {
+				return err
+			}
+
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				resp.Body.Close()
+				return err
+			}
+			resp.Body.Close()
+
+			recs, err := decodeAsCSV(body)
+			if err != nil {
+				return err
+			}
+
+			if firstEvent {
+				if timstamp, ok := rec.Record().Fields()[s.config.EventMonitoringMethod.EventLogFile.Cursor.Field].(string); ok {
+					s.cursor.EventLogFile.FirstEventTime = timstamp
+				}
+			}
+
+			if timstamp, ok := rec.Record().Fields()[s.config.EventMonitoringMethod.EventLogFile.Cursor.Field].(string); ok {
+				s.cursor.EventLogFile.LastEventTime = timstamp
+			}
+
+			for _, val := range recs {
+				jsonStrEvent, err := json.Marshal(val)
+				if err != nil {
+					return err
+				}
+
+				err = publishEvent(s.publisher, s.cursor, jsonStrEvent, "EventLogFile")
+				if err != nil {
+					return err
+				}
+				totalEvents++
+			}
+			firstEvent = false
+		}
+
+		if res.MoreRecords() {
+			res, err = res.Next()
+			if err != nil {
+				return err
+			}
+		} else {
+			break
+		}
+	}
+	s.log.Debugf("Total events: %d", totalEvents)
+
+	return nil
+}
+
+// getSFDCConfig returns a new Salesforce configuration based on the configuration.
+func getSFDCConfig(cfg *config) (*sfdc.Configuration, error) {
+	var (
+		creds *credentials.Credentials
+		err   error
+	)
+
+	if cfg.Auth == nil {
+		return nil, errors.New("no auth provider enabled")
+	}
+
+	switch {
+	case cfg.Auth.OAuth2.JWTBearerFlow != nil && cfg.Auth.OAuth2.JWTBearerFlow.isEnabled():
+		pemBytes, err := os.ReadFile(cfg.Auth.OAuth2.JWTBearerFlow.ClientKeyPath)
+		if err != nil {
+			return nil, fmt.Errorf("problem with client key path for JWT auth: %w", err)
+		}
+
+		signKey, err := jwt.ParseRSAPrivateKeyFromPEM(pemBytes)
+		if err != nil {
+			return nil, fmt.Errorf("problem with client key for JWT auth: %w", err)
+		}
+
+		passCreds := credentials.JwtCredentials{
+			URL:            cfg.Auth.OAuth2.JWTBearerFlow.URL,
+			ClientId:       cfg.Auth.OAuth2.JWTBearerFlow.ClientID,
+			ClientUsername: cfg.Auth.OAuth2.JWTBearerFlow.ClientUsername,
+			ClientKey:      signKey,
+		}
+
+		creds, err = credentials.NewJWTCredentials(passCreds)
+		if err != nil {
+			return nil, fmt.Errorf("problem with credentials: %w", err)
+		}
+	case cfg.Auth.OAuth2.UserPasswordFlow != nil && cfg.Auth.OAuth2.UserPasswordFlow.isEnabled():
+		passCreds := credentials.PasswordCredentials{
+			URL:          cfg.Auth.OAuth2.UserPasswordFlow.TokenURL,
+			Username:     cfg.Auth.OAuth2.UserPasswordFlow.Username,
+			Password:     cfg.Auth.OAuth2.UserPasswordFlow.Password,
+			ClientID:     cfg.Auth.OAuth2.UserPasswordFlow.ClientID,
+			ClientSecret: cfg.Auth.OAuth2.UserPasswordFlow.ClientSecret,
+		}
+
+		creds, err = credentials.NewPasswordCredentials(passCreds)
+		if err != nil {
+			return nil, fmt.Errorf("problem with credentials: %w", err)
```

Comment:
nit: this error message could be more descriptive

Reply:
Kept the prefix message simple and the same across all authentication mechanisms (password,jwt) because the actual valuable information will come from err like what actually happened.

Or you are asking me to use a better prefix?

Reply:
done!